### ISSUE TITLE
Only one of independent rules triggered

### ISSUE THREAD

**Issue by kagayat** [jruizgit/rules#190]
Have a simple ruleset with three independent rules.

```
from durable.lang import *

with ruleset('zservice'):

    # rule A
    @when_all(+m.inputformat)
    def print_input_format(c):
        print('Input format: {0}'.format(c.m.inputformat))

    # rule B
    @when_all(m.inputformat.imatches('text'))
    def print_input(c):
        print('Input: {0}'.format(c.m.input))

    # rule C
    @when_all(+m.eventid)
    def track_eventid(c):
        print('Tracking eventid: {0}'.format(c.m.eventid))

run_all()
```

Posting the following event:

`{"inputformat": "Text", "eventid": "ABC000001", "input": "Nothing to see here"}`

On the event, only one rule (seemingly quasi-random) is triggered. Each defined rule is applicable to the event, as proven by commenting out two other rules and posting event to a single-rule ruleset.

Am I missing something important in how rules are triggered?

**Comment by jruizgit** [comment]
Hi, thanks for asking the question. Posting en event results in only one rule being triggered (events can only be observed once). Asserting a fact should trigger all rules. For a detail description of events and facts, please go to:

https://github.com/jruizgit/rules/blob/master/docs/py/reference.md#basics

By default all rules have the same priority, when that happens the first rule in the sequence will be selected (but this is not guaranteed). You can change the priority of the rules to define which rule is triggered first in cases when the outcome is ambiguous. To learn more about priorities, please go to:

https://github.com/jruizgit/rules/blob/master/docs/py/reference.md#conflict-resolution

**Comment by kagayat** [comment]
Thank you for clarification.
